### DISCUSSION TITLE
Cache estate variables across all requests when environment is selected

### DISCUSSION THREAD

**Discussion by davew94** [hoppscotch/hoppscotch#1698] [Ideas]
If I have an environment selected i want to be able to keep the environment variables cached in the javascript across all my saved requests without the need to go back into Environments > select a different environment >reselect my original environment to re-populate the javascript.

This is ideal when you are running different requests against the same environment.

As a side note > there appears to be a bug where if you have an environment selected that has variables > you open a new collection request, the variables are not populated if you select the same environment from the drop down bar this doesn't populate the javascript. You need to choose a different environment then choose your original environment to populate the JavaScript variables.

**Comment by liyasthomas** [comment]
@davew94 sorry you faced such an issue - we're looking into reproducing the bug.

Also, I guess ability to populate entire environment variables without selection would not be a good idea considering multiple environments will have same variable names. Let us know what exactly is your use case.

**Reply by davew94** [reply]
hi @liyasthomas - i  think this was an issue with my setup as i've not been able to recreate since i created new config.

Please discard this for just now.

**Reply by liyasthomas** [reply]
Thanks for the response
